scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,688 @@
+import * as React from 'react';
+// @ts-ignore
+import ReactCrop from 'react-image-crop';
+import 'react-image-crop/dist/ReactCrop.css';
+import {ISuperdesk, IArticle} from 'superdesk-api';
+import {isEmpty, omit, pick, isEqual, cloneDeep, flatten} from 'lodash';
+
+import {VideoEditorTools} from './VideoEditorTools';
+import {VideoTimeline} from './VideoTimeline/VideoTimeline';
+import {VideoEditorHeader} from './VideoEditorHeader';
+import {VideoEditorThumbnail} from './VideoEditorThumbnail';
+import {IVideoEditor, IThumbnail, ICrop} from './interfaces';
+
+interface IProps {
+    article: IArticle;
+    superdesk: ISuperdesk;
+    onClose: () => void;
+    onArticleUpdate: (articleUpdate: IArticle) => void;
+}
+
+interface IState {
+    isDirty: boolean;
+    trim: {
+        start: number;
+        end: number;
+    };
+    cropImg: string;
+    thumbnails: Array<IThumbnail>;
+    loading: boolean;
+    loadingText: string;
+    scale: number;
```

Comment:
This is used to scale video to fit into container (but not exceed maximum video size) to avoid video [too small](https://user-images.githubusercontent.com/2076953/64971798-817bf280-d8a8-11e9-99db-8d203527e4c9.png) or overflow

Reply:
Couldn't it work CSS only?

Reply:
We have tried it before but couldn't make it to work.
We need to scale video down (or up) when applied rotate transformation. If scale video up, we must ensure that it's not exceeded video size to avoid breaking video images. Do you have any suggestions?

Reply:
I've looked it up and understand it now. When you apply a CSS transform, the parent element doesn't take up the dimensions of the transformed element inside.

Keep it as it is then.